local_vars/local_vars.cc

```diff
@@ -484,8 +492,32 @@ class LocalNameInserter {
 
     void postTransformSend(core::MutableContext ctx, ast::ExpressionPtr &tree) {
         auto &original = ast::cast_tree_nonnull<ast::Send>(tree);
-        if (original.numPosArgs() == 1 && ast::isa_tree<ast::ZSuperArgs>(original.getPosArg(0))) {
-            tree = lowerZSuperArgs(ctx, std::move(tree));
+        // If the super calls occurs inside a block, or a module, we would like to leave it as untyped,
+        // rather than checking it. This code replaces those <super> with <untypedSuper>,
+        // and then dispatchCallSymbol will ignore those.
+        auto newMethodName = core::Names::untypedSuper();
+        if (ctx.state.typedSuper && !scopeStack.back().insideBlock && !scopeStack.back().insideModule) {
+            newMethodName = core::Names::super();
+        }
```

Comment:
I think at least the `insideBlock` check needs to be a little more aggressive, because of code like:

```ruby
foo.class_exec do
  def frob
    super
  end
end
```

(Similarly for `module_exec` and `instance_exec`, I guess?)

The code in its current form will cause Sorbet to expand the `super` call as though it was typed, but we don't have good static information about where that `frob` method is winding up.  Tests would be good here.

I am less sure that the `insideModule` check needs to more aggressively walk the stack, because for:

```ruby
module A
  class << self
    def foo
      super
    end
  end
end
```

I...think that comes out OK, because `foo` is defined on a (singleton) class, not a module?

It also would be slightly nicer, IMHO, to attempt to delay this computation until we decide that we need it, maybe pull the computation of `newMethodName` into a function and call it inside the two blocks below?

Reply:
> I am less sure that the insideModule check needs to more aggressively walk the stack, because for:

yes, in this example, the call to `super` could actually hit a method, if the module `extend`'s something that provides a `foo` method. In that case, it's a singleton class, which is a class. For the same reason that an [abstract singleton method](https://sorbet.org/docs/abstract#abstract-singleton-methods) doesn't make sense, so too is it okay for us to treat these super calls statically.

Good test cases, thanks.